base/db/R/dbfiles.R

```diff
@@ -636,3 +636,63 @@ dbfile.id <- function(type, file, con, hostname=PEcAn.remote::fqdn()) {
     invisible(NA)
   }
 }
+
+#' Function to Insert Records into the Formats and Variables Tables
+#' @name dbfile.format.variable.insert
+#' @param format_name Format Name
+#' @param header TRUE/FALSE Boolean describing whether or not the file contains a header
+#' @param skip The number of lines in the header
+#' @param mimetype_id The mimetype id of the mimetype that is associated with the format
+#' @param format_notes Any useful notes that describe the format being created
+#' @param con Connection object to connect to bety
+#' @param format_variables A TRUE/FALSE Boolean used to toggle a format-variables db query
+#' @param description variable descripthon
+#' @param units variable units
+#' @param var_notes notes to further describe the variable
+#' @param var_name variable name
+#' @param max maximum possible value
+#' @param min minimum possible value
+#' @param standard_name PEcAn specific name? 
+#' @param label ?
+#' @param type ?
+#' @author Liam Burke
+#'
+#' @return formatid
+#' @export
+#' 
+#' @examples
+#' \dontrun{
+#' bety <- betyConnect()
+#'   dbfile.format.variable.insert('Test_Format', header = TRUE, skip = 2, mimetype_id = 1060, con = bety$con, format_variables = FALSE)
+#' }
+dbfile.format.variable.insert <- function(format_name, header, skip, mimetype_id, format_notes = "", con, format_variables = FALSE, 
+                                          description = NA, units = NA, var_notes = NA, var_name = NA, max = NA, min = NA, standard_name = NA, label = NA, type = NA){
+ 
+  formatid <- get.id(
+    table = "formats",
+    colnames = c('mimetype_id', 'name'),
+    values = c(mimetype_id, format_name),
+    con = con,
+    create = TRUE,
+    dates = TRUE)
+
+    #### Formats Query ####
+    cmd <- paste0("INSERT INTO formats ",
+                   "(header, skip, mimetype_id, notes, name) VALUES (",
+                   header, ", ", skip, ", '", mimetype_id, "', '", format_notes, "','", format_name, "')")
+
+    db.query(query = cmd, con = con)
+
+    #### Format Variables Query ####
+    if(format_variables == TRUE){
+    cmd_fv <- paste0("INSERT INTO variables ",
```

Comment:
This query is wrong. First, you need to loop over all the variables (virtually no files will just have one column). Second, you need to be inserting records into `formats_variables` not `variables` as the goal is to associate variables with a specific format. As such you've got to revise all the columns being passed. You also need to use the correct format ID when creating these relationships

Reply:
Got it. Makes sense.